### DISCUSSION TITLE
Future of the Network Map

### DISCUSSION THREAD

**Discussion by Botched1** [zwave-js/zwave-js-ui#1979] [General]
Are there any plans to update/change the network map? On large networks it really isn't very useful as you can either see only small parts of the mesh, or if zoomed out to see the whole thing then you can't read any of the node text.

Example - Here is my network of 98 nodes, not a very useful graph.  A table view may be better for large networks, or maybe a more traditional node drawing like the one in the node-red zwavejs2mqtt nodes.
![image](https://user-images.githubusercontent.com/19985320/142692755-ed676ab8-f533-4acb-8273-1245e5381ad1.png)

**Comment by robertsLando** [comment]
That's the result of another question where everyone has expose their idea of network map, I'm open to suggestions but I would like to have the final solution and not an alternative to it

**Reply by Botched1** [reply]
I think that may be the heart of the problem. Someone with a small Network may love the existing graph, whereas it is not useful for a large network. I'm not 100% sure there is one representation that works for all networks.

**Reply by robertsLando** [reply]
I could create two different implementations of it, we just need to understand what's the best one for large networks.

cc @AlCalzone @kpine @blhoward2

**Reply by AlCalzone** [reply]
I don't know what's the best honestly. Personally I've gone with the below thing in my ioBroker adapter, but for large networks this is equally unreadable.

![grafik](https://user-images.githubusercontent.com/17641229/142829583-b3499c30-c824-479c-9511-9bb10fd451c8.png)

I've thought about a self-organizing representation using a [force-directed graph](https://observablehq.com/@d3/force-directed-graph) but due to the large number of mesh neighbors, this is just going to be a big blob.

**Reply by kpine** [reply]
Maybe this is why the simple matrix topology seems to be popular (PC-Controller, Hubitat, Homeseer, SmartThings?). It communicates what the current graph is really showing, that is just the node neighbors, in a simple fashion. It also scales better for large networks.

![image](https://user-images.githubusercontent.com/791794/142830703-dc90ca7f-b9bd-42ef-9c2d-6061513e6ccb.png)

@robertsLando In case you forgot about it, the same subject was discussed in quite detail in #384.

**Reply by robertsLando** [reply]
Yep exactly, that's the issue I was referencing in my first answer :) Seems that the actual implementation was the most voted but it doesn't fit very well for large networks

**Reply by blhoward2** [reply]
What about just allowing the user to filter by location and other criteria, then display those and their connections? There really isn't a need to view the entire network at once if it's unreadable. 

Or go with a grid...though I find that matrix garish. 

Something like:
Node -> hop -> hop ->controller RSSI etc. in a row and don't try to draw a map at all.

**Reply by AlCalzone** [reply]
I'm still working on it but if we don't show all neighbor relations but rather the routes from and to the controller, the mesh view might be usable.

**Reply by robertsLando** [reply]
I personally like @blhoward2 idea about filtering nodes in the mesh
